src/internet_identity/src/storage.rs

```diff
@@ -1063,20 +1062,45 @@ impl<M: Memory + Clone> Storage<M> {
     /// Used in `update_account` to update an existing account.
     fn update_existing_account(
         &mut self,
-        params: UpdateExistinAccountParams,
-    ) -> Result<AccountNumber, StorageError> {
-        match self.stable_account_memory.get(&params.account_number) {
-            None => Err(StorageError::AccountNotFound {
+        params: UpdateExistingAccountParams,
+    ) -> Result<Account, StorageError> {
+        // Check if account exists for given anchor number, origin and account number
+        let application_number = self.lookup_application_number_with_origin(&params.origin);
+        let account_reference = self
+            .find_account_reference(
+                params.anchor_number,
+                application_number,
+                Some(params.account_number),
+            )
+            .ok_or(StorageError::AccountNotFound {
                 account_number: params.account_number,
-            }),
-            Some(storable_account) => {
-                let mut storable_account = storable_account.clone();
-                storable_account.name = params.name;
-                self.stable_account_memory
-                    .insert(params.account_number, storable_account);
-                Ok(params.account_number)
-            }
-        }
+            })?;
+        // Check if account is an existing account (reserved with an account number)
```

Comment:
It's not checking if is an existing account, it already is, you are in `update_existing_account`. Maybe just check whether it exists.